src/core/toga/app.py

```diff
@@ -16,6 +16,14 @@ class MainWindow(Window):
     def __init__(self, id=None, title=None, position=(100, 100), size=(640, 480), factory=None):
         super().__init__(id=id, title=title, position=position, size=size, factory=factory)
 
+class WindowsRegistry(set):
+
+    def __init__(self, app):
+        self.app = app
+
+    def add(self, window):
+        super().add(window)
+        window.app = self.app
```

Comment:
Since we're overloading `set`, there are a bunch of other methods that could be used to modify the set of windows (`update`, `difference_update`, `intersection_update`, `symmetric_difference_update`,...). While these methods won't have much *practical* use, they *will* appear any API discovery tooling. So, we either need to (a) provide an implementation for these methods, or (b) provide a "set like" API that proxies an actual set internally. 

Personally, I'd lean towards (b).

Reply:
I've implemented b) in 5b08573e8. Has that hit the mark?